packages/common/src/types/filter.ts

```diff
@@ -46,6 +46,7 @@ export interface FilterRule<
     target: T;
     settings?: S;
     disabled?: boolean;
+    required?: boolean;
```

Comment:
Added this extra property, LMK if I should rename to something more specific like: 
`requiredToHaveValue` or similar

Reply:
From this PR: https://github.com/lightdash/lightdash/pull/9490
Seems like there will be a new property `required` that will live in `settings` 

Got suggestions on naming @owlas ?

Reply:
Are these filter rules not also used for saved charts too? Or is this property only being attached to dashboard filters?

It would be confusing to apply this to saved charts if it has no effect

Reply:
Just found a practical example of this. If I "explore from here" I can see that the URL for the saved chart contains a filter with `required: true` and this will get saved onto the chart. What's the expected behaviour here? Could the filter somehow interfere with other dashboard filters later?

We need to type the dashboard filters separately if they have different properties and have an explicit filter conversion when we explore from here.

Reply:
Thanks for pointing this out @owlas, makes no sense to save this onto the chart! I've created a PR to handle the `required` + this filtering before this piece of work here: https://github.com/lightdash/lightdash/pull/9559

Reply:
This has been addressed and we can use `required` in dashboard filters in this PR, now that the above PR has been merged